pennylane/_device.py

```diff
@@ -732,16 +727,20 @@ def batch_transform(self, circuit):
         elif (
             len(circuit._obs_sharing_wires) > 0
             and not hamiltonian_in_obs
-            and not qml.measurements.Sample in return_types
-            and not qml.measurements.Probability in return_types
+            and qml.measurements.Sample not in return_types
+            and qml.measurements.Probability not in return_types
+            and qml.measurements.Counts not in return_types
```

Comment:
This leaves me wondering: should we have a test case for this? I.e., I'd assume there is a test for which we should have had an error, but now it passes, right?

Reply:
Yes sorry, this change was mentioned in [this comment](https://github.com/PennyLaneAI/pennylane/pull/2924#pullrequestreview-1069428448) (at the end). I decided adding it here because it is a related topic. But it actually solved a different bug (I will also mention it in the changelog).

Before:
```python
import pennylane as qml
dev = qml.device("default.qubit", wires=4, shots=1000)
@qml.qnode(dev)
def circuit():
    qml.PauliX(0)
    return qml.expval(qml.PauliX(0)), qml.counts(qml.PauliY(0))
print(circuit())
```
Would return:
```
(tensor(-0.002, requires_grad=True), tensor(-0.004, requires_grad=True))
```
Which makes no sense.

Now it raises:
```
pennylane.QuantumFunctionError: Only observables that are qubit-wise commuting Pauli words can be returned on the same wire
```

Reply:
@AlbertMitjans that's great! :100: Let's have a test case for it in this PR.

Reply:
Test added!